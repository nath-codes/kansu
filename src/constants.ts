export const ANIMATION_MAX_DIMENSION = 75;
export const ANIMATION_OPACITY_SPEED = 0.01;
export const ANIMATION_REPOSITION_SPEED = 0.3;
export const ANIMATION_RESIZE_SPEED = 0.6;
export const ANIMATION_TOTAL_SHAPES = 100;

export const ARIA_EMAIL = "Drop me an email";
export const ARIA_GITHUB = "View my projects on Github";
export const ARIA_INSTAGRAM = "Follow me on Instagram";
export const ARIA_LINKEDIN = "Connect with me on LinkedIn";

export const COLOR_DARK_BLUE_RGBA = [32, 47, 60, 0];
export const COLOR_GREY_RGBA = [73, 85, 94, 0];
export const COLOR_WHITE_RGBA = [255, 255, 255, 0];

export const COLORS = [COLOR_DARK_BLUE_RGBA, COLOR_GREY_RGBA, COLOR_WHITE_RGBA];

export const PAGE_TITLES = {
  ABOUT: "About",
  CLIENTS: "Clients",
  CONTACT: "Contact",
  HOME: "Home",
};

export const META_TITLES = {
  ABOUT: PAGE_TITLES.ABOUT,
  CLIENTS: PAGE_TITLES.CLIENTS,
  CONTACT: PAGE_TITLES.CONTACT,
  HOME: PAGE_TITLES.HOME,
};

export const PAGE_URLS = ["/", "/about", "/clients", "/contact"];

export const NAV = [
  { title: PAGE_TITLES.HOME, url: PAGE_URLS[0] },
  { title: PAGE_TITLES.ABOUT, url: PAGE_URLS[1] },
  { title: PAGE_TITLES.CLIENTS, url: PAGE_URLS[2] },
  { title: PAGE_TITLES.CONTACT, url: PAGE_URLS[3] },
];

export const URL_EMAIL = "mailto:hello@kansu.co";
export const URL_GITHUB = "https://github.com/nathan-kansu";
export const URL_INSTAGRAM = "https://www.instagram.com/nathan.kansu";
export const URL_LINKEDIN = "https://www.linkedin.com/in/nathan-kansu/";
