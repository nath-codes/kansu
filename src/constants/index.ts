export const ANIMATION_MAX_DIMENSION = 75
export const ANIMATION_TOTAL_SHAPES = 100
export const ANIMATION_RESIZE_SPEED = 0.6
export const ANIMATION_OPACITY_SPEED = 0.01
export const ANIMATION_REPOSITION_SPEED = 0.3
export const COLOR_DARK_BLUE_RGBA = [32, 47, 60, 0]
export const COLOR_GREY_RGBA = [73, 85, 94, 0]
export const COLOR_WHITE_RGBA = [255, 255, 255, 0]
export const URL_GITHUB = 'https://github.com/nath-codes'
export const URL_INSTAGRAM = 'https://www.instagram.com/nathan.kansu'
