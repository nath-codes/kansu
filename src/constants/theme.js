export default {
  colors: {
    black: '#14212c',
    grey: '#49555e',
    white: '#ffffff',
  },
  fonts: {
    body: '"Montserrat", sans-serif',
    heading: '"Roboto", sans-serif',
  },
}
