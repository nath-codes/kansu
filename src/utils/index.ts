import { WindowLocation } from 'reach__router'

const getPageNav = (location: WindowLocation | undefined, urls: string[]) => {
  if (location && location.pathname) {
    const currentIndex = urls.findIndex(
      (url: string) => url === location.pathname
    )
    const lastIndex = urls.length - 1
    const nextPage =
      currentIndex === lastIndex ? undefined : urls[currentIndex + 1]
    const prevPage = currentIndex === 0 ? undefined : urls[currentIndex - 1]

    return { nextPage, prevPage }
  }
}

export default getPageNav
