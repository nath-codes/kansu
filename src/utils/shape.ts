import {
  ANIMATION_MAX_DIMENSION,
  COLOR_DARK_BLUE_RGBA,
  COLOR_GREY_RGBA,
  COLOR_WHITE_RGBA,
} from '../constants'

const colors: [number[], number[], number[]] = [
  COLOR_DARK_BLUE_RGBA,
  COLOR_GREY_RGBA,
  COLOR_WHITE_RGBA,
]

const generateDimension = () =>
  Math.round(Math.random() * ANIMATION_MAX_DIMENSION)

const generateFill = () => {
  const fillColorKey = Math.floor(Math.random() * colors.length)
  return colors[fillColorKey]
}

const generatePosition = (limit: number, dimension: number) => {
  const edge = limit - dimension
  return Math.round(Math.random() * edge)
}

export const generateShapeFactory = (
  canvasHeight: number,
  canvasWidth: number
) => {
  const height = generateDimension()
  const width = generateDimension()

  return {
    fill: generateFill(),
    height,
    width,
    x: generatePosition(canvasWidth, width),
    y: generatePosition(canvasHeight, height),
  }
}
