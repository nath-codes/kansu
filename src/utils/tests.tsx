import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { theme } from '../styles/theme'

export const renderWithTheme = (component: JSX.Element) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
