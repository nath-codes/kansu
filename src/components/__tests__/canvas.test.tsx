import 'jest-styled-components'
import React from 'react'
import { Canvas } from '../Canvas'
import { renderWithTheme } from '../../utils/tests'

import { COLOR_DARK_BLUE_RGBA } from '../../constants'

describe('Canvas', () => {
  it('renders correctly', () => {
    const props = {
      height: 100,
      shapes: [
        {
          fill: COLOR_DARK_BLUE_RGBA,
          height: 10,
          width: 10,
          x: 0,
          y: 0,
        },
      ],
      width: 100,
    }

    const tree = renderWithTheme(<Canvas {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
