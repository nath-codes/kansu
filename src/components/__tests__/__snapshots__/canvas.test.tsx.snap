// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canvas renders correctly 1`] = `
<canvas
  className=""
/>
`;
